lib/__test__/query.test.js

```diff
@@ -48,24 +48,68 @@ describe('query', () => {
         y = {type: 'Identifier', name: 'y'}
     })
 
-    suite('getVariableFactorName', query.getVariableFactors, [
+    suite('getVariableFactors', query.getVariableFactors, [
         '2x',
         'x^2',
         '2x^2',
-        '2 x y z'
+        '2 x y z',
     ])
 
     suite('getCoefficientsAndConstants', query.getCoefficientsAndConstants, [
         '2',
         '2x',
-        '2x^2',
+        '2x^2', 
         '2 x y z',
         '2x^3 + 3',
         'x^3 + y^3 + x y z + 3',
         '2/3 + 3 + cos(4)',
-        '3x^2 * 2x^2'
+        '3x^2 * 2x^2',
+        '2/3(x + 1)^1'
     ])
-    
+
+    it('isVariableFactor', () => {
+        assert(query.isVariableFactor(parse('x')))
+        assert(query.isVariableFactor(parse('xyz')))
+        assert(query.isVariableFactor(parse('x^2')))
+        assert(query.isVariableFactor(parse('(x+1)^2')))
+        assert(!query.isVariableFactor(parse('2x^2')))
+        assert(!query.isVariableFactor(parse('x y z')))
+        assert(!query.isVariableFactor(parse('x + y')))
+    })
+
+    it('isPolynomialTerm', () => {
+        assert(query.isPolynomialTerm(parse('x')))
+        assert(query.isPolynomialTerm(parse('2x')))
+        assert(query.isPolynomialTerm(parse('2x^2')))
+        assert(query.isPolynomialTerm(parse('(x+2)^2')))
+        assert(query.isPolynomialTerm(parse('(x+2)^(x+2)')))
+        assert(query.isPolynomialTerm(parse('-2')))
+        assert(!query.isPolynomialTerm(parse('x^-2')))
```

Comment:
I agree.  It says that fractional exponents aren't allowed (but irrational coefficients are).